### PR TITLE
 reorg: modular file reorganization of C source code - v4

### PR DESCRIPTION
**Pull request by jasonish**
Continuation of https://github.com/OISF/suricata/pull/9941.

Changes from last PR:
- Disable clang-format around written out byte arrays. This was a python regex script, it might have got a few things wrong
- Clang-format complete C code base
- Move some more protocols

Some protocols don't fit the current "model" so don't migrate well in the script, that is what I'll work on next (htp, snmp, ...).

### CODE REVIEW

File: src/app-layer-ssh.c [hunk 1]
```diff
@@ -914,10 +932,18 @@ static int SSHParserTest15(void)
     Packet *p = NULL;
 
     uint8_t sshbuf1[] = "SSH-2.0-MySSHClient-0.5.1\r\n";
+    // clang-format off
```
Comment (victorjulien) [inline src/app-layer-ssh.c:941]: can the script be updated to recognize the pattern below? 4 arrays, can just use a single off/on

Comment (jasonish) [reply src/app-layer-ssh.c:941]: Its becoming a gnarly regex.  Probably easier to not apply it to signal line arrays.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16861

**Comment by jasonish** [comment]
Continued at https://github.com/OISF/suricata/pull/9960.

**Pull request closed without merging**